e2e/v2_curl_test.go

```diff
@@ -52,62 +53,117 @@ func testCurlPutGet(t *testing.T, cfg *etcdProcessClusterConfig) {
 	expectGet := `{"action":"get","node":{"key":"/testKey","value":"foo","`
 
 	if cfg.clientTLS == clientTLSAndNonTLS {
-		if err := cURLPut(epc, "testKey", "foo", expectPut); err != nil {
+		if err := cURLPut(epc, "/v2/keys", "testKey", "foo", expectPut, "", "", false); err != nil {
```

Comment:
To test `curl XPUT` requests with different endpoints (`/v2/users`, `/v2/roles`)